# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediacontrol/MediaControlKeysManager.cpp
# Commit: 714b2fb21e23
# Full Hash: 714b2fb21e2341e8e2bf1be1271ca0ed5e526443
# Author: alwu <alwu@mozilla.com>
# Date: 2020-06-09 09:21:34
# Regressor Bug: 1640998
# File Overlap Count: 1
# Description:
#   Bug 1640998 - part5 : set supported keys to the event source r=chunmin
#   
#   This patch will
#   - add a method `SetSupportedMediaKeys()` on `MediaControlKeysEventSource`
#   - set main controller's supported key to the event source
# ==============================================================================

diff -r a48b6554ba9c -r 714b2fb21e23 dom/media/mediacontrol/MediaControlKeysManager.cpp
--- a/dom/media/mediacontrol/MediaControlKeysManager.cpp	Mon Jun 08 22:16:46 2020 +0000
+++ b/dom/media/mediacontrol/MediaControlKeysManager.cpp	Tue Jun 09 02:23:35 2020 +0000
@@ -134,5 +134,17 @@
   }
 }
 
+void MediaControlKeysManager::SetSupportedMediaKeys(
+    const MediaKeysArray& aSupportedKeys) {
+  mSupportedKeys.Clear();
+  for (const auto& key : aSupportedKeys) {
+    LOG_INFO("Supported keys=%s", ToMediaControlKeysEventStr(key));
+    mSupportedKeys.AppendElement(key);
+  }
+  if (mEventSource && mEventSource->IsOpened()) {
+    mEventSource->SetSupportedMediaKeys(mSupportedKeys);
+  }
+}
+
 }  // namespace dom
 }  // namespace mozilla